### PR TITLE
dev-util/hip: bump version to 4.2.0

### PR DESCRIPTION
**Pull request by littlewu2508**
Update hip-3.5.1-config-cmake-in.patch to 4.2.0
lpl_ca removed, so hip-3.9.0-lpl_ca-add-include.patch not used

Package-Manager: Portage-3.0.20, Repoman-3.0.3
Signed-off-by: Yiyang Wu <xgreenlandforwyy@gmail.com>

### CODE REVIEW

**Comment by heroxbd** [comment]
Thanks for the pull request!

A new hip-4.2.0-config-cmake-in.patch is added.  Please add some
explanations in it referring to

  https://wiki.gentoo.org/wiki/User:Veremit/Patch_format

Benda

**Comment by littlewu2508** [comment]
@heroxbd  hip-4.2.0 ebuild is ready. Sandbox violation issue fixed, and some cleaning work in hip-4.1.0 and hip-4.1.0-r1 is inherited as well. Please review the changes, thanks.

File: dev-util/hip/files/hip-4.2.0-config-cmake-in.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
+diff --color -uprN orig/hip-config.cmake.in HIP-rocm-4.2.0/hip-config.cmake.in
```
Comment (heroxbd) [inline dev-util/hip/files/hip-4.2.0-config-cmake-in.patch:1]: I know this is adopted from hip-3.5.1-config-cmake-in.patch, but please also add a patch header to explain the logic behind it.

**Review by heroxbd — CHANGES_REQUESTED** [review]

**Comment by thesamesam** [comment]
Note that the CI is red on the PR so the issue with dev-util/roctracer is now in the tree too. Could you have a look at it please? Thanks!

**Comment by littlewu2508** [comment]
> Note that the CI is red on the PR so the issue with dev-util/roctracer is now in the tree too. Could you have a look at it please? Thanks!

There is a circular dependency in hip and roctracer: roctracer depend on hip, while hip[profile] depend on roctracer. In my local development branch I've upgraded both versions and pass the QA check. So should I push roctracer-4.2 together in this PR?

**Comment by heroxbd** [comment]
> Note that the CI is red on the PR so the issue with dev-util/roctracer is now in the tree too. Could you have a look at it please? Thanks!

Sorry I have missed the CI report.  Will roll out a solution ASAP

**Pull request closed without merging**